File: src/talk.py
import os
import time
import mm2
import textwrap
import scripts

def talk():
    save = scripts.savePull()
    var6 = save[5]
    var7 = save[6]
    var8 = save[7]
    scripts.screenClear()
    scripts.versionHeader()
    scripts.invDisplay()
    if var6 == 0 or var7 == 0 or var8 == 1: #Normal/post-explain
        print("Your friends are relieved that you're alright.\n")
    if var6 == 1 and var8 == 0: #Murder, pre-explain
        print('Your friends look at you with horrified expressions.\n')
    if var7 == 1 and var6 == 0 and var8 == 0: #Murdered before reset
            print('...they seem worried.\n')
    if var6 == 0 or var8 == 1: #Normal/post-explain
        print('CHIT-CHAT [1]')
    if var6 == 1 and var8 == 0 and var7 == 1: #Murder
        print("WHAT'S WITH THAT LOOK? [1]")
    print('STARE [2]')
    print('BACK [3]')
    while True:
        try:
            choose = int(input('\nACTION >> '))
            if choose == 1:
                if var6 == 1 and var8 == 0:
                    talkSel1C()
                else:
                    talkSel1()
            if choose == 2:
                talkSel2()
            if choose == 3:
                if var6 == 1:
                    mm2.mainMenu2()
                if var6 == 0:
                    print('\nYou tell your friends you need a minute.')
                    time.sleep(2)
                    mm2.mainMenu2()
            if choose > 4 or choose < 0:
                scripts.inpErrorHandler()
                talk()
        except ValueError:
            scripts.inpErrorHandler()
            talk()

def talkSel1A():
    save = scripts.savePull()
    var7 = save[6]
    var8 = save[7]
    if var7 == 0 or var8 == 1:
        scripts.screenClear()
        scripts.versionHeader()
        scripts.invDisplay()
        print("Your friends ask you why you didn't use any of your tools to contact them.\n")
        print('UH... [1]')
        print('BACK [2]')
        while True:
            try:
                choose = int(input('\nACTION >> '))
                if choose == 1:
                    print("\nYour friends tell you to not to worry about it.")
                    time.sleep(2)
                    talk()
                if choose == 2:
                    print("\nYou pretend that you're getting a call on your phone to avoid this awkward conversation.")
                    time.sleep(4)
                    talk()
                if choose > 2 or choose < 0:
                    scripts.inpErrorHandler()
                    talkSel1()
            except ValueError:
                scripts.inpErrorHandler()
                talkSel1()
    if var7 == 1 and var8 == 0:
        talkSel1B()

def talkSel1B():
    print('\nYour friends ask you a question.')
    time.sleep(2)
    print('"Liam, is everything alright?"\n')
    time.sleep(2)
    print('EXPLAIN [1]')
    print('BACK [2]')
    while True:
        try:
            choose = int(input('\nACTION >> '))
            if choose == 1:
                line_ext = 7
                state_ext = 1
                scripts.saveWriter(line_ext, state_ext)
                talk()
                print("\nYou explain to them that you're just feeling off.")
                time.sleep(3)
                print("\n:)")
                talk()
            if choose == 2:
                print("\nThey wouldn't get it.")
                time.sleep(5)
                talk()
            if choose > 2 or choose < 0:
                scripts.inpErrorHandler()
                talkSel1B()
        except ValueError:
            scripts.inpErrorHandler()
            talkSel1B()

def talkSel1C():
    save = scripts.savePull()
    var6 = save[5]
    scripts.screenClear()
    scripts.versionHeader()
    scripts.invDisplay()
    if var6 == 1:
        print('Your friends barely stammer out a question.')
        time.sleep(2)
        print('"Liam, why are you covered in blood?"\n')
        time.sleep(2)
        print('EXPLAIN [1]')
        print('RUN AWAY [2]')
        while True:
            try:
                choose = int(input('\nACTION >> '))
                if choose == 1:
                    print('\nYou explain to your friends that it was self-defense.')
                    time.sleep(2)
                    print('They understand, and help you clean off.')
                    time.sleep(2)
                    print('They seriously bought that?')
                    time.sleep(0.5)
                    talkSel1CG()
                if choose == 2:
                    print("\nYou can't face them.")
                    time.sleep(2)
                    mm2.mainMenu2()
                if choose > 2 or choose < 0:
                    scripts.inpErrorHandler()
                    talkSel1C()
            except ValueError:
                scripts.inpErrorHandler()
                talkSel1C()
    if var6 == 0:
        talk()

def talkSel1():
    save = scripts.savePull()
    var3 = save[2]
    var7 = save[6]
    var8 = save[7]
    if var3 == 1:
        if var7 == 0 or var8 == 1:
            print('')
            print(textwrap.fill("Your friends mentioned that they heard your scanner signal, and knew to come find you.", 75))
            time.sleep(5)
            talk()
        if var7 == 1 and var8 == 0:
            talkSel1B()
    if var3 == 0:
        talkSel1A()

def talkSel2():
        scripts.screenClear()
        scripts.versionHeader()
        scripts.invDisplay()
        print("You just...stare at them. They look bewildered.\n")
        print('KEEP STARING [1]')
        print('GO BACK [2]')
        while True:
            try:
                choose = int(input('\nACTION >> '))
                if choose == 1:
                    print('')
                    print(textwrap.fill("Your unblinking eyes eventually cause them to wonder if there's something seriously wrong with you.", 75))
                    time.sleep(5)
                    talk()
                if choose == 2:
                    print("\nYeah, this is pretty bizarre.")
                    time.sleep(1.5)
                    talk()
                if choose > 2 or choose < 0:
                    scripts.inpErrorHandler()
                    talkSel2()
            except ValueError:
                scripts.inpErrorHandler()
                talkSel2()

File: src/talk.rb
require './handy'
require './mm2'

def talk
  begin
    var6 = saveRead[5]
    var7 = saveRead[6]
    var8 = saveRead[7]
    screenClear
    versionHeader
    invDisplay
    if var6 == 0 or var7 == 0 or var8 == 1
      puts "\nYour friends are relieved that you're alright."
    end
    if var6 == 1 and var8 == 0
      puts "\nYour friends look at you, horrified."
    end
    if var7 == 1 and var6 == 0 and var8 == 0
      puts "\nThey seem worried."
    end
    if var6 == 0 or var8 == 1
      puts "\nCHAT"
    end
    if var6 == 1 and var8 == 0 or var7 == 1
      puts "\nWHAT'S WITH THAT LOOK? [1]"
    end
    puts "STARE [2]"
    puts "BACK [3]"
    print "\nACTION >> "
    choose = Integer(gets.chomp)
    if choose == 1
      if var6 == 1 and var8 == 0
        talksel1C
      else
        talkSel1
      end
    end
    if choose == 2
      talkSel2
    end
    if choose == 3
      if var6 == 1
        mainMenu2
      end
      if var6 == 0
        puts "\nYou tell your friends you need a minute."
        sleep(2)
        mainMenu2
      end
      mainMenu2
    end
    if choose > 3 or choose < 1
      inpErHandler
      talk
    end
  rescue ArgumentError
    inpErrorHandler
    talk
  end
end

def talkSel1
  begin
    screenClear
    versionHeader
    invDisplay
    var3 = saveRead[2]
    var7 = saveRead[6]
    var8 = saveRead[7]
    if var3 == 1
      if var7 == 0 or var8 == 1
        puts "\nYour friends mentioned that your scanner had painful latency."
        sleep(4)
        talk
      end
      if var7 == 1 and var8 == 0
        talksel1B
      end
    end
  if var3 == 0
    talksel1A
  end
  rescue ArgumentError
    inpErHandler
    talkSel1
  end
end

def talkSel1A
  begin
    screenClear
    versionHeader
    invDisplay
    puts "\nYour friends ask you why you didn't use any of your tools to contact them."
    puts "\nUH... [1]"
    puts "BACK [2]"
    print "\nACTION >> "
    choose = Integer(gets.chomp)
    if choose == 1
      puts "\nYour friends tell you not to worry...but to do it next time."
      sleep(4)
      talk
    end
    if choose == 2
      puts "\nYou pretend you're getting a call on your phone to avoid this awkward conversation."
      sleep(4)
      talk
    end
    if choose > 2 or choose < 1
      inpErHandler
      talkSel1A
    end
  rescue ArgumentError
    inpErHandler
    talkSel1A
  end
end

def talkSel1B
  begin
    puts "\nYour friends ask you a question."
    sleep(2)
    puts "Liam, are you alright?"
    sleep(2)
    puts "EXPLAIN [1]"
    puts "BACK [2]"
    print "\nACTION >> "
    choose = Integer(gets.chomp)
    if choose == 1
      line_ext = 7
      state_ext = 1
      saveWrite(line_ext, state_ext)
      puts "\nYou explain to them that you're just feeling off."
      sleep(3)
      puts "\n:)"
      sleep(0.04)
      talk
    end
    if choose == 2
      puts "\nThey wouldn't get it."
      sleep(4)
      talk
    end
    if choose > 2 or choose < 1
      inpErHandler
      talkSel1B
    end
  rescue ArgumentError
    inpErHandler
    talkSel1B
  end
end

def talkSel1C
  begin
    var6 = saveRead[5]
    screenClear
    versionHeader
    invDisplay
    puts "\nYour friends barely stammer out a question."
    sleep(2)
    puts "Liam, why are you covered in blood?"
    sleep(3)
    puts "\nEXPLAIN [1]"
    puts "RUN AWAY [2]"
    print "\nACTION >> "
    choose = Integer(gets.chomp)
    if choose == 1
      puts "\nYou tell your friends it was self-defense."
      sleep(2)
      puts "They seem to understand."
      sleep(4)
      talk
    end
    if choose == 2
      puts "\nYou can't face them."
      sleep(2)
      mainMenu2
    end
    if choose > 2 or choose < 1
      inpErHandler
      talkSel1C
    end
  rescue ArgumentError
    inpErHandler
    talkSel1C
  end
end

def talkSel2
  begin
    screenClear
    versionHeader
    invDisplay
    puts "\nYou just...stare at them. They look bewildered."
    puts "\nKEEP STARING [1]"
    puts "GO BACK [2]"
    print "\nACTION >> "
    choose = Integer(gets.chomp)
    if choose == 1
      puts "\nYour unblinking eyes eventually cause them to wonder if there's something seriously wrong with you."
      sleep(4)
      talk
    end
    if choose == 2
      puts "\nYeah, this is pretty bizarre."
      sleep(1.5)
      talk
    end
    if choose > 2 or choose < 1
      inpErHandler
      talkSel2
    end
  rescue ArgumentError
    inpErHandler
    talkSel2
  end
end
